lightning/src/ln/channelmanager.rs

```diff
@@ -131,6 +132,13 @@ pub(super) enum PendingHTLCStatus {
 	Fail(HTLCFailureMsg),
 }
 
+pub(super) struct PendingInterceptedHTLC {
```

Comment:
Yeah, my bad.  agreed we should rename it if so.

Reply:
Looking at this now I'm a little confused.  Does HTLCForwardInfo::AddHTLC need to "hold" this struct? They are literally exactly the same already.  Was this not the case back in July when this was first written?

Unless I'm misremembering something couldn't I just get rid of this PendingInterceptedHTLC struct entirely and just store (or clone) the entire HTLCForwardInfo as the intercepted htlc instead of essentially copying it into this new struct?

Reply:
I guess it would mean matching on HTLCForwardInfo anywhere I need to use it and ignoring/erroring if we came across a ::FailHTLC when we expected an ::AddHTLC.

Do you know what you meant by hold it when it's already the same data?

Reply:
Oh whoops.  There's a separate comment thread in here about this.  You meant change it to HTLCForwardInfo::AddHTLC(RenamedPendingInterceptedHTLC).

Is the reason to update the AddHTLC variant to hold a struct instead of using named fields so that when I need to store the "pending intercepted htlcs" I can take (or clone) this inner struct instead of having to store the entire HTLCForwardInfo?  Thus also eliminating the need to match against it.

Reply:
So I starting working on this and it's a bit of a pain so I wanted to make sure it still makes sense to do this before I continue.

A couple things I wanted to highlight or questions I have about it if we do want to move forward with that idea:

1) There's a lot of places where you match on HTLCForwardInfo::AddHTLC { fields...} and do a nested match on the PendingRoutingHTLC::Forward variant at the same time.  I was doing this instead in the new version:

```
// destructuring to avoid having to write `pending_add_htlc_info.<field_name>` everywhere
let PendingAddHTLCInfo {
  prev_short_channel_id,
  prev_htlc_id,
  forward_info: PendingHTLCInfo {
    routing,
    incoming_shared_secret,
    payment_hash,
    amt_to_forward,
    outgoing_cltv_value,
    ..
  },
  prev_funding_outpoint
} = pending_add_htlc_info;

// instead of the nested match, putting the whole arm inside this if let instead :/
if let PendingHTLCRouting::Forward { onion_packet, .. } = routing {
  ...
}
```

Is there a cleaner way to do this?

2) How can I insert the PendingAddHTLCInfo (what I renamed it for now) into my map without cloning it? Is it possible? Is it okay if I have to clone it to insert it?  Otherwise it's fields get used / thus moved before I need to insert.

3) Is there an issue with serialization / backwards compat? The old enums will be serialized with the ::AddHTLC struct variant and the new version will want to deserialize with this tuple / nested struct version.  Is this okay?

Reply:
Ok sounds good, just wanted to make sure it made sense before continuing down this path. Thanks!